test/index.test.js

```diff
@@ -0,0 +1,178 @@
+
+/**
+ * Module dependencies.
+ */
+
+import { ValidationError } from 'objection';
+import clearDatabase from './utils/clear-database';
+import modelFactory from './utils/model-factory';
+
+/**
+ * Test `FoobarController`.
+ */
+
+describe('FoobarController', () => {
+  beforeEach(clearDatabase);
+
+  describe('$beforeInsert', () => {
+    it('should throw an error if there is no fields or identifiers options.`', async () => {
+      const TestModel = modelFactory();
+
+      try {
+        await TestModel.query().insert({});
+
+        fail();
+      } catch (e) {
+        expect(e.message).toBe('Fields and identifiers options must be defined.');
+      }
+    });
+
+    it('should throw a `ValidationError` with the unique fields that are already used.`', async () => {
+      const TestModel = modelFactory({
+        fields: ['bar', 'foo']
+      });
+
+      await TestModel.query().insert({ bar: 'bar', foo: 'foo' });
+
+      try {
+        await TestModel.query().insert({ bar: 'bar', foo: 'foo' });
+
+        fail();
+      } catch (e) {
+        expect(e).toBeInstanceOf(ValidationError);
+        expect(e.data).toEqual({
+          bar: [{
+            keyword: 'unique',
+            message: 'bar already in use.'
+          }],
+          foo: [{
+            keyword: 'unique',
+            message: 'foo already in use.'
+          }]
+        });
+      }
+    });
+
+    it('should insert the given data ignoring null values validation.', async () => {
+      const TestModel = modelFactory({
+        fields: ['bar', 'foo'],
+        ignoreNulls: true
+      });
+
+      await TestModel.query().insert({ bar: 'bar', biz: 'biz', foo: null });
+
+      const data = { bar: 'buz', foo: null };
+      const { id } = await TestModel.query().insert(data);
+      const test = await TestModel.query().findById(id);
+
+      expect(test).toEqual({ ...data, biz: null, id });
+    });
+
+    it('should insert the given data.', async () => {
+      const TestModel = modelFactory({
+        fields: ['bar', 'foo']
+      });
+
+      const data = { bar: 'bar', biz: 'biz', foo: 'foo' };
```

Comment:
WDYT about passing this directly as argument. Also, in the other tests like this one.